application_test.go

```diff
@@ -427,3 +428,21 @@ func TestApplicationConfiguration(t *testing.T) {
 	assert.True(t, ok)
 	assert.Equal(t, ErrCodeNotFound, apiErr.ErrCode)
 }
+
+func TestWaitOnApplication(t *testing.T) {
+	endpoint := newFakeMarathonEndpoint(t, nil)
+	defer endpoint.Close()
+
+	err := endpoint.Client.WaitOnApplication(fakeAppName, 0*time.Millisecond)
```

Comment:
@timoreimann 
Yep, my bad.
The better way is to extract anonymous function from WaitOnApplication method into separate one, so that we can isolate logic we want to test from side effects of deadline method.